functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -63,9 +62,64 @@ func (p *HelmChartInflationGeneratorPlugin) establishTmpDir() (err error) {
 	return err
 }
 
+func (p *HelmChartInflationGeneratorPlugin) ConfigureAuth(items []*fn.KubeObject) (err error) {
+	if p.Auth == nil {
+		return nil
+	}
+	if p.Auth.GetKind() != "Secret" {
+		return fmt.Errorf("auth `kind` must be `Secret`")
+	}
+
+	var targetSecret *fn.KubeObject
+	for _, i := range items {
+		iNamespace := i.GetNamespace()
+		if iNamespace == "" {
+			iNamespace = "default"
+		}
+		authNamespace := p.Auth.Namespace
+		if authNamespace == "" {
+			authNamespace = "default"
+		}
+		if i.GetKind() == "Secret" && i.GetName() == p.Auth.Name && iNamespace == authNamespace {
+			targetSecret = i
+		}
+	}
+	if targetSecret == nil {
+		return fmt.Errorf("could not find Secret %q identified by auth", p.Auth)
+	}
+
+	u, found, err := targetSecret.NestedString("data", "username")
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("could not find username in Secret")
+	}
+
+	pass, found, err := targetSecret.NestedString("data", "password")
+	if err != nil {
+		return err
+	}
```

Comment:
Another format can be using the `As` method.

Reply:
then you can call the `corev1.Secret` functions directly. It might be easier for other SWE to understand.
FYI, error happens when the type of `data/password` path is not a string. I guess the main reason you use `NestedString` but not `NestedStringOrDie` is for `found`. But how about `pass` value is ""? Shall it give the same user message "secret not given" or acceptable?

Reply:
Hm. yeah I think empty `pass` should be an error.

I updated the logic to use `As` and `corev1.Secret`.